### PR TITLE
Columns for open and closed match count

### PR DESCRIPTION
**Pull request by gigxz**
Is this going to be super slow in prod? If so we could pre calculate the counts per contact in the controllers

https://boston-cas.dev.test/contacts
https://boston-cas.dev.test/admin/users

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -100,7 +100,6 @@ Layout/DotPosition:
     - 'app/models/cas_seeds/client_shelter_agency_contact.rb'
     - 'app/models/cas_seeds/opportunity_housing_subsidy_admin_contact.rb'
     - 'app/models/concerns/related_default_contacts.rb'
-    - 'app/models/contact.rb'
```
Comment (eanders) [inline .rubocop_todo.yml]: Pretty sure this didn't get fixed.  Let's leave the todos here unless we fix them.

File: app/models/contact.rb [hunk 2]
```diff
@@ -98,4 +98,16 @@ def self.contact_type_for(input_name)
   def phone_for_display
     phone.presence || cell_phone
   end
+
+  def client_opportunity_matches
+    @client_opportunity_matches ||= client_opportunity_match_contacts.joins(:match)
+  end
```
Comment (eanders) [inline app/models/contact.rb]: This is equivalent to the relation: `client_opportunity_matches` above.  This is overriding the ActiveRecord method.  If we need different behavior, I'd suggest something like:
```suggestion
  has_many :matches, through: :client_opportunity_match_contacts, source: :match
```

File: app/models/contact.rb [hunk 3]
```diff
@@ -98,4 +98,16 @@ def self.contact_type_for(input_name)
...
+
+  def open_match_count
+    client_opportunity_matches.merge(ClientOpportunityMatch.open).count
+  end
+
+  def closed_match_count
+    client_opportunity_matches.merge(ClientOpportunityMatch.closed).count
+  end
```
Comment (eanders) [inline app/models/contact.rb]: These are going to introduce N+1, so we should see if we can batch calculate.  I think something in the controller is probably easiest.  This should happen after the pagination.
```
@open_matches = Contact.where(id: @contacts.map(&:id)).joins(:matches).merge(ClientOpportunityMatch.open).group(:id).count
@closed_matches = Contact.where(id: @contacts.map(&:id)).joins(:matches).merge(ClientOpportunityMatch.closed).group(:id).count
```
Those would then be keyed on contact ID.

**Review by eanders — CHANGES_REQUESTED** [review]
See inline for some thoughts.

File: app/models/contact.rb (hunk 3 above)
Comment (gigxz) [reply app/models/contact.rb]: thanks, updated to use this

**Pull request merged**